Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -376,6 +369,54 @@ CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* compo
 
 #pragma region WIC_HELPERS
 
+DisplayTexture* _CGImageGetDisplayTexture(CGImageRef image) {
+    RETURN_NULL_IF(!image);
+    CGIWICBitmap* iwicImg = dynamic_cast<CGIWICBitmap*>(image->ImageSource().Get());
+    RETURN_NULL_IF(!iwicImg);
+    return iwicImg->GetDisplayTexture()->GetTexture();
+}
+
+// Return the data pointer to the Image data.
+void* _CGImageObtainImageBytes(CGImageRef image) {
+    RETURN_NULL_IF(!image);
+    return image->Data();
+}
+
+CGImageRef _CGImageCreateCustomWICImage(IDisplayTexture* texture, WICPixelFormatGUID pixelFormat, UINT height, UINT width) {
+    RETURN_NULL_IF(!texture);
+
+    ComPtr<IWICBitmap> bitmap = Make<CGIWICBitmap>(texture, pixelFormat, height, width);
+    RETURN_NULL_IF(!bitmap);
+
+    CGImageRef imageRef = __CGImage::CreateInstance();
+    imageRef->SetImageSource(bitmap);
+
+    return imageRef;
+}
+
+CGImageRef _CGImageConvertImageToPixelFormat(CGImageRef image, WICPixelFormatGUID pixelFormat) {
+    RETURN_NULL_IF(!image);
+    if (IsEqualGUID(image->PixelFormat(), pixelFormat)) {
+        CGImageRetain(image);
```

Comment:
Depends on the context. They can be mutable or immutable.
If you access the pixels via a lock, yes they can be mutable. Given this is off the code of CALayer processing, it's expected this way.

---
In reply to: [87049935](https://github.com/Microsoft/WinObjC/pull/1332#discussion_r87049935) [](ancestors = 87049935)